File: Languages/PHP/Day011.php
<!-- Languages - PHP - Day 11 -->
<!-- Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables. -->

<!-- Create a function that takes two numbers as parameters, calculates their product, and returns the result. Call this function and print the result.-->

<?php

// Function to calculate the product of two numbers
function product($num1, $num2) {
    return $num1 * $num2;
}

// Function to get a valid number from the user
function getValidNumber($prompt) {
    // Loop until a valid number is entered
    while (true) {
        echo $prompt;
        $input = readline();
        if (is_numeric($input)) {
            return (float)$input;
        } else {
            // If the input is not a number, print an error message
            echo "Invalid input. Please enter a valid number.\n";
        }
    }
}

// Get two numbers from the user
$num1 = getValidNumber("Enter the first number: ");
$num2 = getValidNumber("Enter the second number: ");

// Calculate and print the product of the two numbers
echo "The product of the two numbers is: " . product($num1, $num2) . "\n";

?>

File: Languages/Rust/Day011.rs
// Languages - Rust - Day 11
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that takes two numbers as parameters, calculates their product, and returns the result. Call this function and print the result.

use std::io;

// Function to calculate the product of two numbers
fn product(num1: i32, num2: i32) -> i32 {
    num1 * num2
}

fn get_input(prompt: &str) -> i32 {
    // Loop until a valid number is entered
    loop {
        println!("{}", prompt);
        let mut input = String::new();
        // Read input from the user
        io::stdin().read_line(&mut input).expect("Failed to read input");
        match input.trim().parse::<i32>() {
            Ok(num) => return num,
            // Handle invalid input
            Err(_) => println!("Invalid input. Please enter a valid number."),
        }
    }
}

fn main() {
    // Get input from the user
    let num1 = get_input("Enter the first number: ");
    let num2 = get_input("Enter the second number: ");

    // Calculate the product of the two numbers
    let result = product(num1, num2);
    // Print the result
    println!("The product of {} and {} is: {}", num1, num2, result);
}
